### PR TITLE
Cleanup: Random Seed in Tests

### PR DESCRIPTION
**Pull request by JakobEliasWagner**
# Cleanup: Random Seed in Tests

## Description

This pr introduces a fixture in `conftest.py` that automatically sets the random seed for `random` and `torch`.

### Which issue does this PR tackle?

  - Currently the random seed is set in files, which can be inconsistent.

### How does it solve the problem?

  - Introduces a fixture in `conftest.py` that sets the random seed.
  - This enhances the reproducibility of all tests (though system and system dependencies still matter).

### How are the changes tested?

  - All tests still run without errors.

## Checklist for Contributors

- [x] Scope: This PR tackles exactly one problem.
- [x] Conventions: The branch follows the `feature/title-slug` convention.
- [x] Conventions: The PR title follows the `Bugfix: Title` convention.
- [x] Coding style: The code passes all pre-commit hooks.
- [x] Documentation: All changes are well-documented.
- [x] Tests: New features are tested and all tests pass successfully.
- [x] Changelog: Updated CHANGELOG.md for new features or breaking changes.
- [x] Review: A suitable reviewer has been assigned.

## Checklist for Reviewers:

- [x] The PR solves the issue it claims to solve and only this one.
- [x] Changes are tested sufficiently and all tests pass.
- [x] Documentation is complete and well-written.
- [x] Changelog has been updated, if necessary.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+
 pytest_plugins = [
     "tests.transforms.fixtures",
 ]
+
+
+@pytest.fixture(autouse=True)
+def set_random_seed():
+    random.seed(0)
+    torch.manual_seed(0)
```
Comment (samuelburbulla) [inline tests/conftest.py:13]: Add `np.random.seed(0)`

**Review by samuelburbulla — CHANGES_REQUESTED** [review]
Great solution! Let's also set numpy's random seed.

File: tests/conftest.py (hunk 1 above)
Comment (JakobEliasWagner) [reply tests/conftest.py:13]: `numpy` is currently a secondary dependence (not linked to continuity directly right now). `tensorboard`, `contuourpy`, `matplotlib`, `optuna`, and `pandas` depend on it however. I will add this call and add `numpy` as a dependency of `continuity` for consistency.

Comment (JakobEliasWagner) [reply tests/conftest.py:14]: Adapted to this in [71d231f](https://github.com/aai-institute/continuity/pull/66/commits/71d231fdf606d927e2fe29180593f90ebc1e5fe6) and [7935e0c](https://github.com/aai-institute/continuity/pull/66/commits/7935e0c0a5d9b6da76a62292f2eb5334534e3e54).

**Review by samuelburbulla — APPROVED** [review]

**Pull request merged**